Python/import.c

```diff
@@ -1018,14 +1018,14 @@ module_dict_for_exec(PyThreadState *tstate, PyObject *name)
     /* If the module is being reloaded, we get the old module back
        and re-use its dict to exec the new code. */
     d = PyModule_GetDict(m);
-    if (_PyDict_GetItemIdWithError(d, &PyId___builtins__) == NULL) {
-        if (_PyErr_Occurred(tstate) ||
-            _PyDict_SetItemId(d, &PyId___builtins__,
-                              PyEval_GetBuiltins()) != 0)
-        {
-            remove_module(tstate, name);
-            return NULL;
-        }
+    int r = _PyDict_ContainsId(d, &PyId___builtins__);
+    if (r == 0) {
+        r = _PyDict_SetItemId(d, &PyId___builtins__,
+                              PyEval_GetBuiltins());
```

Comment:
`PyEval_GetBuiltins()` is cheap function so we can use `PyDict_SetDefault()` here, and `builtin_eval_impl`, `builtin_exec_impl` if you want to.

Reply:
There is no `_PyDict_SetDefaultId()` and I do not want to introduce it just for these three cases. Using `PyDict_SetDefault()` + `_PyUnicode_FromId()` will not make large difference:
```c
    int r = _PyDict_ContainsId(d, &PyId___builtins__);
    if (r == 0) {
        r = _PyDict_SetItemId(d, &PyId___builtins__,
                              PyEval_GetBuiltins());
    }
    if (r < 0) {
        remove_module(tstate, name);
        return NULL;
    }
```
vs
```c
    PyObject *tmp = _PyUnicode_FromId(&PyId___builtins__);
    if (tmp != NULL) {
        tmp = PyDict_SetDefault(d, tmp, PyEval_GetBuiltins());
    }
    if (tmp == NULL) {
        remove_module(tstate, name);
        return NULL;
    }
```